stdlib/public/core/StringBridge.swift

```diff
@@ -537,6 +596,213 @@ internal func _NSStringFromUTF8(_ s: UnsafePointer<UInt8>, _ len: Int)
   )._bridgeToObjectiveCImpl()
 }
 
+@_fixed_layout
+@usableFromInline
+@objc internal final class _SwiftNSMutableString :
+  _SwiftNativeNSMutableString, _AbstractStringStorage {
+  private var _contents: String
+  
+  internal init(_ str: String) {
+    _contents = str
+    super.init()
+    assert(_contents._guts.isFastUTF8)
```

Comment:
`_internalInvariant`

If you have multiple conditions you'd like to check, you can follow the check pattern:

```swift
#if !INTERNAL_CHECKS_ENABLED
  @inline(__always)
  internal func _invariantCheck() {}
#else
  internal func _invariantCheck() {
    ...
  }
#endif // INTERNAL_CHECKS_ENABLED
```

These can find all kinds of assumption violations early